src/insert.rs

```diff
@@ -279,9 +266,66 @@ impl<T> Insert<T> {
         Ok(mem::replace(&mut self.buffer, BytesMut::with_capacity(BUFFER_SIZE)).freeze())
     }
 
+    fn init_request(&mut self) -> Result<()> {
```

Comment:
Let's mark it `#[cold] #[inline(never)]` in order to forbid inlining into `do_write()`.

```
group                    master                                 query-level                            query-level-cold
-----                    ------                                 -----------                            ----------------
insert/lz4               1.12     62.5±3.62ns   854.2 MB/sec    1.07     59.5±3.17ns   896.8 MB/sec    1.00     55.6±3.25ns   960.6 MB/sec
insert/lz4hc(4)          1.08     64.8±5.12ns   824.7 MB/sec    1.04     62.3±5.50ns   857.1 MB/sec    1.00     59.8±4.27ns   893.2 MB/sec
insert/no compression    1.05     59.9±3.80ns   891.7 MB/sec    1.03     58.3±5.40ns   916.4 MB/sec    1.00     56.8±4.68ns   940.4 MB/sec
```